### DISCUSSION TITLE
Unable to access Raspberry Pi Pico, COM ports unavailable

### DISCUSSION THREAD

**Discussion by dylan-berndt** [micropython/micropython#10257] [RP2040 / Pico] [answered]
I have been trying to program a raspberrry pi pico through USB serial, however, recently I can't access the board through any COM port. I have tried closing down every application and process that could be blocking usage of the port, switching from Thonny to Pycharm, resetting the board, and tried accessing the board from my laptop, but no matter what I do I keep getting the error "could not open port 'COM7': PermissionError(13, "Access Denied", None, 5).

I was able to program the board for a short while, but unplugging it and plugging it back in at one point has completely prevented that. If anybody has any idea what I could do to fix this, it would be greatly appreciated.

**Comment by robert-hh** [comment]
Do you always connect the board to the same physical USB port at your PC? If you plug the board to a different port, Windows may assign a different COM number.
Do you connect directly to the PC or through an external USB hub? I have the situation with RPi boards and an external hub, that they will sometimes not be detected, or only after 40 seconds delay.

**Reply by dylan-berndt** [reply]
Directly to PC and I've used several different ports, they always somehow show as COM7 in the device manager, and reassigning the COM number doesn't change anything.

**Comment by dhalbert** [comment]
"Permission Denied" can happen if some other application already has the port open. Check that you don't have a terminal program open on the same port, etc.

You can clean up unused ports using Uwe Sieber's USB Cleanup program. Details here: https://learn.adafruit.com/welcome-to-circuitpython/troubleshooting#device-errors-or-problems-on-windows-3094694. I don't think that's the problem here, but when you start getting into double-digit COM ports, that tool can help clean things up.

**Reply by dylan-berndt** [reply]
I removed the device but upon reconnecting, it assigned the device to COM3, which shows as a bluetooth link in device manager. I still can't connect using either Thonny, Pycharm, or PuTTy, and Thonny lists a separate COM3 for the bluetooth link and a "n/a" COM3 underneath it which it can't access.

**Comment by dylan-berndt** [comment, marked as answer]
Turns out I had a main.py file still hidden on the flash memory that was crashing the board somehow. Installing Circuitpython wipes that memory so I just had to install that on the board after using BOOTSEL, and then using BOOTSEL again so I could reinstall Micropython.
